src/tests/tpch/sb_tpch.c

```diff
@@ -0,0 +1,404 @@
+        log_text(LOG_FATAL, "Invalid value of root-path, got NULL.");
+        return 1;
+    }
+
+    tpch.size = (unsigned int)size;
+    tpch.root_path = root_path;
+    return 0;
+}
+
+static char *add_path_to_root(char *add)
+{
+    int size = strlen(tpch.root_path) + strlen(add) + 2;
+    char *path = malloc(sizeof(char) * size);
+
+    if (path == NULL) {
+        return NULL;
+    }
+    strcat(path, tpch.root_path);
+    if (strlen(tpch.root_path) > 0 && tpch.root_path[strlen(tpch.root_path) - 1] != '/') {
+        strcat(path, "/");
+    }
+    strcat(path, add);
+    return path;
+}
+
+static char** get_script_arguments(const char *path)
+{
+    const int nb_args = 4;
+    char args_str[nb_args][32] = {
+            "--size",
+            "",
+            "--mysql-params",
+            "tpch"
+    };
+    char **args = malloc(sizeof(char *) * (nb_args + 2));
+    if (args == NULL)
+        return NULL;
+
+    sprintf(args_str[1], "%d", tpch.size);
+
+    args[0] = malloc(sizeof(char) * strlen(path));
+    if (args[0] == NULL) {
+        free(args);
+        return NULL;
+    }
+
+    strcpy(args[0], path);
+    for (int i = 0; i < nb_args; i++) {
+        args[i+1] = malloc(sizeof(char) * strlen(args_str[i]));
+        if (args[i+1] == NULL) {
+            for (int j = i; j >= 0; j--)
+                free(args[j]);
+            free(args);
+            return NULL;
+        }
+        strcpy(args[i+1], args_str[i]);
+    }
+    args[nb_args+1] = NULL;
+    return args;
+}
+
+static int execute_init_script(void)
+{
+    char *path = NULL;
+    char *exec_path = NULL;
+    char **args = NULL;
+    int status = 0;
+    pid_t pid = fork();
+
+    if (pid == -1) {
+        return 1;
+    }
+    if (pid == 0) {
+        path = add_path_to_root("src/tests/tpch/scripts");
+        if (path == NULL)
+            return 1;
+        chdir(path);
+
+        exec_path = malloc(sizeof(char) * (strlen(path) + strlen("/tpch_init.sh")));
+        if (exec_path == NULL)
+            return 1;
+        strcat(exec_path, path);
+        strcat(exec_path, "/tpch_init.sh");
+
+        args = get_script_arguments(exec_path);
+        if (args == NULL)
+            return 1;
+        int exit_code = execve(exec_path, args, sb_globals.env);
+        if (exit_code == -1)
+            printf("Something went wrong: %s\n", strerror(errno));
+        exit(exit_code);
+    }
+    waitpid(pid, &status, 0);
+
+    free(path);
+    free(exec_path);
+
+    if (WIFEXITED(status))
+        return WEXITSTATUS(status);
+    return 0;
+}
+
+static char *get_sql_file_content(unsigned int id)
+{
+    FILE *f = NULL;
+    long file_size = 0;
+    char *content = NULL;
+    char *file_path = malloc(sizeof(char) * (strlen(tpch.query_path) + strlen("/00.sql")));
+
+    if (file_path == NULL)
+        return NULL;
+    if (id < 10)
+        sprintf(file_path, "%s/0%d.sql", tpch.query_path, id);
+    else
+        sprintf(file_path, "%s/%d.sql", tpch.query_path, id);
```

Comment:
You don't need a conditional here. Printf syntax lets you pad with zeroes just fine. `sprintf(file_path, "%s/%02d.sql", tpch.query_path, id);`